torch/csrc/jit/codegen/cuda/compute_at_map.cpp

```diff
@@ -239,17 +287,127 @@ void ComputeAtMap::mapIds(IterDomain* id0, IterDomain* id1) {
+  std::unordered_set<IterDomain*> count_one_concrete_dims;
+
+  // Should be run on consumer id after producer id and consumer id are mapped
+  // together, will place all id's in consumer_id's disjoint set into
+  // count_one_concrete_dims if consumer_id is in
+  // rfactor_concrete_count_reset_domains
+  auto maybePropagateConcreteCountOne = [&](IterDomain* consumer_id) {
+    if (rfactor_concrete_count_reset_domains.find(consumer_id) ==
+        rfactor_concrete_count_reset_domains.end()) {
+      return;
+    }
+
+    // We need to add the disjoint set associated with id to
+    // count_one_concrete_dims
+    auto disjoint_set_it = disjoint_iter_set_maps_.find(consumer_id);
+    if (disjoint_set_it == disjoint_iter_set_maps_.end()) {
+      // Nothing to propagate to
+      return;
+    }
+    count_one_concrete_dims.insert(
+        disjoint_set_it->second->begin(), disjoint_set_it->second->end());
+  };
+
+  // Detects id's that need to be added to rfactor_concrete_count_reset_domains
+  // from tv and adds them, returns true if anything was added.
+  auto pullConcreteCountResetIds = [&rfactor_concrete_count_reset_domains,
+                                    &count_one_concrete_dims,
+                                    &trivial_reduction_info](TensorView* tv) {
+    bool is_view_like_rfactor = false;
+    // If consumer is not used in any other expression we don't have to
+    // worry about resolving its view like rfactor in any of the maps.
+    if (tv->uses().size()) {
+      // Check if this consumer has an rfactor domain without any
+      // reductions, this would make it a view-like operation
+      if (tv->hasRFactor() &&
+          std::none_of(
+              tv->domain()->domain().begin(),
+              tv->domain()->domain().end(),
+              [&trivial_reduction_info](IterDomain* id) {
+                return id->isRFactorProduct() &&
+                    // If one of the rfactor domains is a reduction we don't
+                    // have to perform special handling.
+                    (id->isReduction() &&
+                     !trivial_reduction_info->isDerived(id));
+              })) {
+        // Definitely rfactor like iteration domain add rfactor iteration
+        // domains to the rfactor_concrete_count_reset_domains set
+
+        for (auto id : tv->domain()->domain()) {
+          if (id->isRFactorProduct()) {
+            // Add id as a domain to reset concrete count on for map propagation
+            rfactor_concrete_count_reset_domains.emplace(id);
+            // Also make sure the id is in count_one_concrete_dims, in case id
+            // never hits a maybePropagateConcreteCountOne
+            count_one_concrete_dims.emplace(id);
+          }
+        }
+
+        is_view_like_rfactor = true;
+      }
+    }
+    return is_view_like_rfactor;
+  };
+
+  auto fusion_inputs = fusion->inputs();
+  // Pull view like rfactor id's out of inputs for later processing.
+  // TODO: Is this needed?
```

Comment:
Yes, the fusion inputs shouldn't have any rfactor iterDomains right now. However, didn't we want to allow that in the future?

Reply:
OK, I didn't know we wanted to allow it. Anyway, these are input tensors, so I don't think it's necessary to reset their counts, though that won't hurt.

Reply:
I think it would be good to generally allow rfactor iter domains on inputs. Maybe it's not a really critical requirement, but I generally disliked resetting the input domains on a fusion.